framework/src/meshmodifiers/ElementSubdomainModifierBase.C

```diff
@@ -0,0 +1,632 @@
+    if (_moving_boundary_names.count(bnd))
+      bnd_info.remove_node(node_id, bnd);
+
+  // Keep track of ghost element changes
+  std::unordered_map<processor_id_type,
+                     std::vector<std::tuple<dof_id_type, unsigned short, BoundaryID>>>
+      add_ghost_sides, remove_ghost_sides;
+
+  // Remove element sides from moving boundaries
+  for (const auto & [elem_id, sides] : _remove_element_sides)
+    for (const auto & [side, bnd] : sides)
+      bnd_info.remove_side(mesh.elemPtr(elem_id), side, bnd);
+
+  // Remove neighbor sides from moving boundaries
+  for (const auto & [elem_id, sides] : _remove_neighbor_sides)
+  {
+    auto elem = mesh.elemPtr(elem_id);
+    for (const auto & [side, bnd] : sides)
+    {
+      bnd_info.remove_side(elem, side, bnd);
+      // Keep track of changes to ghosted elements
+      if (elem->processor_id() != processor_id())
+        remove_ghost_sides[elem->processor_id()].push_back({elem_id, side, bnd});
+    }
+  }
+
+  // Add element sides to moving boundaries
+  for (const auto & [elem_id, sides] : _add_element_sides)
+    for (const auto & [side, bnd] : sides)
+      bnd_info.add_side(mesh.elemPtr(elem_id), side, bnd);
+
+  // Add neighbor sides to moving boundaries
+  for (const auto & [elem_id, sides] : _add_neighbor_sides)
+  {
+    auto elem = mesh.elemPtr(elem_id);
+    for (const auto & [side, bnd] : sides)
+    {
+      bnd_info.add_side(elem, side, bnd);
+      // Keep track of changes to ghosted elements
+      if (elem->processor_id() != processor_id())
+        add_ghost_sides[elem->processor_id()].push_back({elem_id, side, bnd});
+    }
+  }
+
+  Parallel::push_parallel_vector_data(
+      bnd_info.comm(),
+      add_ghost_sides,
+      [&mesh,
+       &bnd_info](processor_id_type,
+                  const std::vector<std::tuple<dof_id_type, unsigned short, BoundaryID>> & received)
+      {
+        for (const auto & [elem_id, side, bnd] : received)
+          bnd_info.add_side(mesh.elemPtr(elem_id), side, bnd);
+      });
+
+  Parallel::push_parallel_vector_data(
+      bnd_info.comm(),
+      remove_ghost_sides,
+      [&mesh,
+       &bnd_info](processor_id_type,
+                  const std::vector<std::tuple<dof_id_type, unsigned short, BoundaryID>> & received)
+      {
+        for (const auto & [elem_id, side, bnd] : received)
+          bnd_info.remove_side(mesh.elemPtr(elem_id), side, bnd);
+      });
+
+  bnd_info.parallel_sync_side_ids();
+  bnd_info.parallel_sync_node_ids();
+  mesh.update();
+}
+
+void
+ElementSubdomainModifierBase::meshChanged()
+{
+  // Clear cached ranges
+  _reinitialized_elem_range.reset();
+  _reinitialized_displaced_elem_range.reset();
+  _reinitialized_bnd_node_range.reset();
+  _reinitialized_displaced_bnd_node_range.reset();
+
+  removeInactiveMovingBoundary(_mesh);
+  if (_displaced_mesh)
+    removeInactiveMovingBoundary(*_displaced_mesh);
+}
+
+void
+ElementSubdomainModifierBase::removeInactiveMovingBoundary(MooseMesh & mesh)
```

Comment:
Two comments here: 
1. This is not just removing inactive moving boundary. Looks like this functions also checks active sides in the elements' family and update the boundary accordingly. So a different name would be better?
2. If a boundary does not contain any sides in one step, the boundary ID still exist in the mesh. So in later steps new sides can be added to this boundary if needed, right? Do you happen to know how this case is handled during output and visualization ( in Exodus and Paraview)?

Reply:
1. A different name would be better. This is specifically for updating the boundary for adaptive mesh refinement, as the boundaries need to be updated from the parent to the children elements. This will be changed to `updateAMRMovingBoundary` to better reflect its purpose. If you have a better idea, that would also be good.
2. That is a good point. I will add a test case to check that new sides can still be added to the boundary.

Reply:
I like `updateAMRMovingBoundary` :) and thanks for agreeing to check that special case

Reply:
Hello, I have added a test case `return_moving_boundary`. The boundary is defined between subdomains 1 and 2. Subdomain 1 changes to 3 then back to 1, which means the boundary does not contain any sides in a timestep, but does return at the later timestep. This looks fine in the Exodus and Paraview. Let me know if you were thinking of a different test.

Reply:
Yes, that is exactly what I meant. Thanks for adding the test!